elastic/kibana
# [Fleet] Fix side effects accross spaces

## Summary

Resolve https://github.com/elastic/kibana/issues/190010

We have a bunch of place in Fleet where when you edit a global settings (fleet server hosts, settings, output), all the related agent policies need to be updated.
That PR fix that to work with Fleet space aware and updated the related agent policies accross all spaces

## Automated tests

I added api integration test to cover those scenarios

## Manual tests

You can manually test that feature with the feature flag `useSpaceAwareness` and enabling the space awareness feature with the following API call:
```
 curl -u elastic:changeme -XPOST "http://localhost:5601/internal/fleet/enable_space_awareness" -H "kbn-xsrf: reporting" -H 'elastic-api-version: 1'
 ```
 
Then you can create policies accross multiple space and assign them a specific download source.
When you delete that download source all the policies should have been updated to use the default download source.

You can test the same for other global settings
